pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -1006,14 +1085,21 @@ func (ctx *context) getServiceBlock(i *OOBInvitation) (*did.Service, error) {
 
 		err = decoder.Decode(svc)
 		if err != nil {
-			return nil, fmt.Errorf("failed to decode service block : %w", err)
+			// for DIDCommV2, decoder.Decode(svc) doesn't support serviceEndpoint as []interface{} representing an array
+			// for model.Endpoint. Manually build the endpoint here in this case.
+			if strings.Contains(err.Error(), "'serviceEndpoint' expected a map, got 'slice'") {
+				extractDIDCommV2EndpointIntoService(svc, s)
```

Comment:
you are passing s by value you need to pass it by reference because you are changing the s struct inside the method you passing